src/main/scala/org/codefeedr/pipeline/buffer/serialization/schema_exposure/RedisSchemaExposer.scala

```diff
@@ -0,0 +1,105 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package org.codefeedr.pipeline.buffer.serialization.schema_exposure
+
+import java.net.URI
+
+import com.redis.RedisClient
+import org.apache.avro.Schema
+
+/**
+  * Exposes (Avro) schema's to Redis.
+  * @param host the server host of Redis.
+  * @param root the root location of the schema's
+  *             DEFAULT: /codefeedr:schemas
+  */
+class RedisSchemaExposer(host: String, root: String = "codefeedr:schemas") extends SchemaExposer {
+
+  private var connection: RedisClient = _
+
+  //connect to the redis client
+  connect()
+
+  /**
+    * Connect with the RedisClient.
+    */
+  private def connect() = {
+    val uri = new URI(host)
+    connection = new RedisClient(uri)
+  }
+
+  /**
+    * Stores a schema bound to a subject.
+    *
+    * @param schema the schema belonging to that topic.
+    * @param subject the subject belonging to that schema.
+    * @return true if correctly saved
+    */
+  override def putSchema(schema: Schema, subject: String): Boolean = {
+    connection.set(s"$root:$subject", schema.toString(true))
+  }
+
+  /**
+    * Get a schema based on a subject.
+    *
+    * @param subject the subject the schema belongs to.
+    * @return None if no schema is found or an invalid schema. Otherwise it returns the schema.
+    */
+  override def getSchema(subject: String): Option[Schema] = {
+    val schemaString = connection.get[String](s"$root:$subject")
+
+    //if no string is found, return None
+    if (schemaString.isEmpty) return None
+
+    //parse the string into a Schema
+    parseSchema(schemaString.get)
+  }
+
+
+  /**
+    * Deletes a schema.
+    *
+    * @param subject the subject the schema belongs to.
+    * @return true if successfully deleted, otherwise false.
+    */
+  override def delSchema(subject: String): Boolean = {
+    val deleted = connection.del(s"$root:$subject")
+
+    //if deleted doesnt return anything or less than 1 key is deleted, return false
+    if (deleted.isEmpty || deleted.get < 1) {
+      return false
+    }
+
+    true
+  }
+
+  /**
+    * Deletes all schemas 'under' the root.
+    */
+  override def deleteAllSchemas() = {
```

Comment:
deleteAll.

The class is already about Schemas, why bother with putting that everywhere?